x/ethbridge/handler.go

```diff
@@ -67,17 +70,55 @@ func handleMsgCreateEthBridgeClaim(ctx sdk.Context, cdc *codec.Codec,
 	return sdk.Result{Events: ctx.EventManager().Events()}
 }
 
-func processSuccessfulClaim(ctx sdk.Context, bankKeeper bank.Keeper, claim string) sdk.Error {
+func processSuccessfulClaim(ctx sdk.Context, supplyKeeper supply.Keeper, claim string) sdk.Error {
 	oracleClaim, err := types.CreateOracleClaimFromOracleString(claim)
 	if err != nil {
 		return err
 	}
 
 	receiverAddress := oracleClaim.CosmosReceiver
-	_, err = bankKeeper.AddCoins(ctx, receiverAddress, oracleClaim.Amount)
+	err = supplyKeeper.MintCoins(ctx, ModuleName, oracleClaim.Amount)
 	if err != nil {
 		return err
 	}
+	err = supplyKeeper.SendCoinsFromModuleToAccount(ctx, ModuleName, receiverAddress, oracleClaim.Amount)
+	if err != nil {
+		panic(err)
+	}
 
 	return nil
 }
+
+func handleMsgBurn(ctx sdk.Context, cdc *codec.Codec,
+	supplyKeeper supply.Keeper, accountKeeper auth.AccountKeeper, msg MsgBurn,
+	codespace sdk.CodespaceType) sdk.Result {
+	account := accountKeeper.GetAccount(ctx, msg.CosmosSender)
+	if account == nil {
+		return sdk.ErrInvalidAddress(msg.CosmosSender.String()).Result()
+	}
+	err := supplyKeeper.SendCoinsFromAccountToModule(ctx, msg.CosmosSender, ModuleName, msg.Amount)
```

Comment:
ditto, this should be called using the interface of the supply module passed to `ethbridge.Keeper`. Check the distribution module for reference

Reply:
will refactor to use expected_keepers and to create a bridge keeper in follow up PR